internal/api/plugin/tree/tree_test.go

```diff
@@ -0,0 +1,307 @@
+// Copyright 2025 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tree
+
+import (
+	"testing"
+
+	"github.com/perses/perses/pkg/model/api/v1/plugin"
+	"github.com/stretchr/testify/assert"
+)
+
+type treeParameter struct {
+	name     string
+	metadata plugin.ModuleMetadata
+	instance any
+}
+
+func TestTree_Add(t *testing.T) {
+	testSuite := []struct {
+		name          string
+		basedTree     Tree[any]
```

Comment:
```suggestion
		baseTree     Tree[any]
```

unless you mean the tree is based 🤓